<?xml version="1.0" ?>
<!-- =================================================================================== -->
<!-- |    This document was autogenerated by xacro from mobman.xacro                   | -->
<!-- |    EDITING THIS FILE BY HAND IS NOT RECOMMENDED                                 | -->
<!-- =================================================================================== -->
<robot name="mobman" xmlns:xacro="http://www.ros.org/wiki/xacro">
  <link name="base">
    <visual>
      <geometry>
        <cylinder length="0.3" radius="0.25"/>
      </geometry>
      <origin xyz="0 0 0.15"/>
      <material name="yello">
        <color rgba="0 1 1 1"/>
      </material>
    </visual>
  </link>
  <link name="body">
    <visual>
      <geometry>
        <box size="0.1 0.1 0.425"/>
      </geometry>
      <origin xyz="0 0 0.2"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <link name="chest">
    <visual>
      <geometry>
        <!---     <cylinder radius="${body_size_x/2-gap}" length="${body_size_z-gap}" /> -->
        <sphere radius="0.2"/>
      </geometry>
      <!--- <origin xyz ="0 0 ${body_size_z/2}" /> -->
      <origin xyz="0 0 0"/>
      <material name="white">
        <color rgba="1 1 1 1"/>
      </material>
    </visual>
  </link>
  <link name="head_x"/>
  <link name="head_z">
    <visual>
      <geometry>
        <box size="0.4 0.3 0.15"/>
      </geometry>
      <origin xyz="0 0 0.1"/>
      <material name="blue">
        <color rgba="1 0 1 1"/>
      </material>
    </visual>
  </link>
  <link name="leftArmUp_x"/>
  <link name="leftArmUp_y"/>
  <link name="leftArmUp_z">
    <visual>
      <geometry>
        <box size="0.1 0.1 0.25"/>
      </geometry>
      <origin xyz="0 0 0.15"/>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
  </link>
  <link name="leftArmDown_x"/>
  <link name="leftArmDown_z">
    <visual>
      <geometry>
        <box size="0.1 0.1 0.2"/>
      </geometry>
      <origin xyz="0 0 0.125"/>
      <material name="green">
        <color rgba="0 1 0 1"/>
      </material>
    </visual>
  </link>
  <link name="leftHand">
    <visual>
      <geometry>
        <box size="0.07 0.07 0.1"/>
      </geometry>
      <origin xyz="0 0 0.075"/>
      <material name="hand">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>
  <link name="rightArmUp_x"/>
  <link name="rightArmUp_y"/>
  <link name="rightArmUp_z">
    <visual>
      <geometry>
        <box size="0.1 0.1 0.25"/>
      </geometry>
      <origin xyz="0 0 0.15"/>
      <material name="red">
        <color rgba="1 0 0 1"/>
      </material>
    </visual>
  </link>
  <link name="rightArmDown_x"/>
  <link name="rightArmDown_z">
    <visual>
      <geometry>
        <box size="0.1 0.1 0.2"/>
      </geometry>
      <origin xyz="0 0 0.125"/>
      <material name="green">
        <color rgba="0 1 0 1"/>
      </material>
    </visual>
  </link>
  <link name="rightHand">
    <visual>
      <geometry>
        <box size="0.07 0.07 0.1"/>
      </geometry>
      <origin xyz="0 0 0.075"/>
      <material name="hand">
        <color rgba="1 1 0 1"/>
      </material>
    </visual>
  </link>
  <joint name="joint_body" type="fixed">
    <parent link="base"/>
    <child link="body"/>
    <origin rpy="0 0 0" xyz="0 0 0.3"/>
  </joint>
  <joint name="joint_chest" type="fixed">
    <parent link="body"/>
    <child link="chest"/>
    <origin rpy="0 0 0" xyz=" 0 0 0.6"/>
  </joint>
  <joint name="joint_head_x" type="continuous">
    <parent link="chest"/>
    <child link="head_x"/>
    <origin rpy="0 0 0" xyz="0 0 0.2"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="joint_head_z" type="continuous">
    <parent link="head_x"/>
    <child link="head_z"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_leftArmUp_x" type="continuous">
    <parent link="chest"/>
    <child link="leftArmUp_x"/>
    <origin rpy="0 1.57 0" xyz="0.2 0 0.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="joint_leftArmUp_y" type="continuous">
    <parent link="leftArmUp_x"/>
    <child link="leftArmUp_y"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="joint_leftArmUp_z" type="continuous">
    <parent link="leftArmUp_y"/>
    <child link="leftArmUp_z"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_leftArmDown_x" type="continuous">
    <parent link="leftArmUp_z"/>
    <child link="leftArmDown_x"/>
    <origin rpy="0 0 0" xyz="0 0 0.3"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="joint_leftArmDown_z" type="continuous">
    <parent link="leftArmDown_x"/>
    <child link="leftArmDown_z"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_leftHand_x" type="continuous">
    <parent link="leftArmDown_z"/>
    <child link="leftHand"/>
    <origin rpy="0 0 0" xyz="0 0 0.25"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="joint_rightArmUp_x" type="continuous">
    <parent link="chest"/>
    <child link="rightArmUp_x"/>
    <origin rpy="0 -1.57 0" xyz="-0.2 0 0.0"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="joint_rightArmUp_y" type="continuous">
    <parent link="rightArmUp_x"/>
    <child link="rightArmUp_y"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 1 0"/>
  </joint>
  <joint name="joint_rightArmUp_z" type="continuous">
    <parent link="rightArmUp_y"/>
    <child link="rightArmUp_z"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_rightArmDown_x" type="continuous">
    <parent link="rightArmUp_z"/>
    <child link="rightArmDown_x"/>
    <origin rpy="0 0 0" xyz="0 0 0.3"/>
    <axis xyz="1 0 0"/>
  </joint>
  <joint name="joint_rightArmDown_z" type="continuous">
    <parent link="rightArmDown_x"/>
    <child link="rightArmDown_z"/>
    <origin rpy="0 0 0" xyz="0 0 0"/>
    <axis xyz="0 0 1"/>
  </joint>
  <joint name="joint_rightHand_x" type="continuous">
    <parent link="rightArmDown_z"/>
    <child link="rightHand"/>
    <origin rpy="0 0 0" xyz="0 0 0.25"/>
    <axis xyz="1 0 0"/>
  </joint>
</robot>

